code/modules/games/cas.dm

```diff
@@ -0,0 +1,120 @@
+// CARDS AGAINST SPESS
+// This is a parody of Cards Against Humanity (https://en.wikipedia.org/wiki/Cards_Against_Humanity)
+// which is licensed under CC BY-NC-SA 2.0, the full text of which can be found at the following URL:
+// https://creativecommons.org/licenses/by-nc-sa/2.0/legalcode
+// Original code by Zuhayr, Polaris Station, ported with modifications
+
+/obj/item/toy/cards/deck/cas
+	name = "\improper CAS deck (white)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is the white deck."
+	icon = 'icons/obj/toy.dmi'
+	icon_state = "deck_caswhite_full"
+	deckstyle = "caswhite"
+	var/blanks = 0
+	var/decksize = 100
+	var/card_text_file = "strings/cas_white.txt"
+	var/list/allcards = list()
+
+/obj/item/toy/cards/deck/cas/blanks
+	name = "\improper CAS deck (white blanks)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is a white deck which contains blanks to be used in custom games of CAS."
+	decksize = 0
+	blanks = 50
+
+/obj/item/toy/cards/deck/cas/black
+	name = "\improper CAS deck (black)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is the black deck."
+	icon_state = "deck_casblack_full"
+	deckstyle = "casblack"
+	blanks = 0
+	decksize = 50
+	card_text_file = "strings/cas_black.txt"
+
+/obj/item/toy/cards/deck/cas/black/blanks
+	name = "\improper CAS deck (black blanks)"
+	desc = "A deck for the game Cards Against Spess, still popular after all these centuries. Warning: may include traces of broken fourth wall.\nThis is a black deck which contains blanks to be used in custom games of CAS."
+	decksize = 0
+	blanks = 50
+
+/obj/item/toy/cards/deck/cas/New()
+	allcards = file2list(card_text_file)
+	var/list/possiblecards = allcards.Copy()
+	if(possiblecards.len < decksize) // sanity check
+		decksize = (possiblecards.len - 1)
+	var/list/randomcards = list()
+	while (randomcards.len < decksize)
+		randomcards += pick_n_take(possiblecards)
+	for(var/i=1 to randomcards.len)
```

Comment:
for(var/i in 1 to randomcards.len), it's faster process wise,just a minor thing tho